src/utils.rs

```diff
@@ -38,6 +40,56 @@ pub async fn run_one_off_command<W>(
     }
 }
 
+pub async fn gc_partitions(context: Arc<DefaultSeafowlContext>) {
```

Comment:
We could also run this on every `DROP TABLE` statement, though it may be an overkill. Alternatively, `sqlparser` supports the [`PURGE`](https://github.com/sqlparser-rs/sqlparser-rs/blob/main/src/ast/mod.rs#L1069) keyword for the `DROP` statement, which can be re-purposed here to initiate GC.

I also like the idea of having this be callable via a command; `sqlparser` doesn't support `VACUUM`, and the potential existing alternatives (i.e. `DEALLOCATE ALL`, `DISCARD ALL`) are not adequate I think, so we'd need to have that customized.

Reply:
As agreed, I've added the `VACUUM PARTITIONS` and `VACUUM TABLES` statement support for now.